src/webui/www/public/scripts/login.js

```diff
@@ -51,10 +51,13 @@ const submitLoginForm = (event) => {
         })
         .then(async (response) => {
                 const responseText = await response.text();
-                if (response.ok && (responseText === "Ok."))
+                if (response.ok && (responseText === "Ok.")) {
                     location.replace(location); // redirect
-                else
+                    location.reload(true);
```

Comment:
For the record, now the browser will send 2 GET requests.
The first `replace()` will be aborted quickly and the second `reload()` will have effect. IMO it is not ideal to have 2 requests but if #20442 proposed it as a fix (and maybe tested it) then we don't have any choice.

Also the `reload()` is a must. If the url contains fragment component (`#fragment`), the first `replace()` will have no effect and `reload()` will still work.